mindsdb/integrations/handlers/merlion_handler/merlion_handler.py

```diff
@@ -0,0 +1,223 @@
+    column_names = list(df.columns.values)
+    mapping = {}
+    back_mapping = {}
+    for column_name in column_names:
+        if column_name.find("`") == -1:
+            continue
+        formatted_name = column_name.replace("`", "")
+        mapping[column_name] = formatted_name
+        back_mapping[formatted_name] = column_name
+    return mapping, back_mapping
+
+
+class MerlionHandler(BaseMLEngine):
+    name = 'merlion'
+
+    ARG_USING_TASK = "task"
+    ARG_USING_MODEL_TYPE = "model_type"
+    ARG_USING_TIME_COLUMN = "time_column"
+    ARG_HORIZON = "horizon"
+    ARG_WINDOW = "window"
+    ARG_TARGET = "target" # only be used to persist args to args.json
+    ARG_COLUMN_SEQUENCE = "column_sequence"  # only be used to persist args to args.json
+
+    KWARGS_DF = "df"
+
+    DEFAULT_MODEL_TYPE = "default"
+    DEFAULT_MAX_PREDICT_STEP = 100
+    DEFAULT_PREDICT_BASE_WINDOW = 10
+
+    PERSISIT_MODEL_FILE_NAME = "merlion_model"
+    PERSISIT_ARGS_KEY_IN_JSON_STORAGE = "args"
+
+    def create(self, target, args=None, **kwargs):
+        df: pd.DataFrame = kwargs.get(self.KWARGS_DF, None)
+        column_name_mapping, column_name_back_mapping = get_column_name_format_mapping(df)
+        df.rename(columns=column_name_mapping, inplace=True)
+
+        # prepare arguments
+        task = args.get(self.ARG_USING_TASK, TaskType.forecast.name)
+        model_type = args.get(self.ARG_USING_MODEL_TYPE, self.DEFAULT_MODEL_TYPE)
+        time_column = args.get(self.ARG_USING_TIME_COLUMN, None)
+        horizon = args.get(self.ARG_HORIZON, self.DEFAULT_MAX_PREDICT_STEP)
+        window = args.get(self.ARG_WINDOW, self.DEFAULT_PREDICT_BASE_WINDOW)
+        # update args for default value maybe has been used, only time column will be set afterwards
+        args[self.ARG_TARGET] = target
+        args[self.ARG_USING_TASK] = task
+        args[self.ARG_USING_MODEL_TYPE] = model_type
+        args[self.ARG_HORIZON] = horizon
+        args[self.ARG_WINDOW] = window
+
+        # check df
+        if df is None:
+            raise Exception("missing required key in args: " + self.KWARGS_DF)
+        else:
+            column_sequence = list(df.columns.values)
+            sorted(column_sequence)
+            df = df[column_sequence]
+            args[self.ARG_COLUMN_SEQUENCE] = column_sequence
+
+        # check task, model_type and get the adapter_class
+        adapter_class = self.__args_to_adapter_class(task=task, model_type=model_type)
+        task_enum = TaskType[task]
+
+        # check and cast to ts dataframe
+        time_column = to_ts_dataframe(df=df, time_col=time_column)
+        args[self.ARG_USING_TIME_COLUMN] = time_column
+
+        # train model
+        model_args = {}
+        if task_enum == TaskType.forecast:
+            model_args[MerlionArguments.max_forecast_steps.value] = horizon
+        adapter: BaseMerlionForecastAdapter = adapter_class(**model_args)
+        print("Training model, args: " + json.dumps(args))
```

Comment:
Best to redirect all `print`s to the mindsdb logger

Reply:
> Best to redirect all `print`s to the mindsdb logger

all "prints" have been replaced by "mindsdb.log"